src/patternfly/components/Drawer/drawer.scss

```diff
@@ -417,7 +412,20 @@ $pf-c-drawer__panel--list--width: (25, 33, 50, 66, 75, 100);
         height: 0;
       }
 
+      flex-direction: var(--pf-c-drawer__panel--m-resizable--FlexDirection);
       padding-left: var(--pf-c-drawer__panel--m-resizable--PaddingLeft);
+
+      > .pf-c-drawer__splitter {
+        right: var(--pf-c-drawer__panel--m-resizable--PaddingLeft);
+        flex-grow: 0;
+        flex-shrink: 0;
+      }
+
+      > .pf-c-drawer__panel-main {
+        flex-shrink: 1;
+        flex-grow: 1;
+        margin-left: calc(var(--pf-c-drawer__panel--m-resizable--PaddingLeft) * -1);
```

Comment:
A couple of things about this - first, just to note about syntax and structure, we typically don't do `calc()`s with a variable that do not belong to the property in the declaration in the CSS. We try as much as we can to do the calcs and refrences to other variables as the value of another variable that matches the property name in a declaration. As an example, for this we would do....

```
--pf-c-drawer__panel--m-resizable__panel-main--MarginLeft: calc(var(--pf-c-drawer__panel--m-resizable--PaddingLeft) * -1);

[...]

margin-left: var(--pf-c-drawer__panel--m-resizable__panel-main--MarginLeft:);
```

However, I don't think the `margin-left` here is necessary, or the `margin-right` on the left panel or `margin-top` on the bottom panel. It looks like you can remove the `padding-left`, `padding-right`, and `padding-top` from the default panel, left panel, and bottom panel now. That padding was used previously to create space for the absolutely positioned splitter element.

And since you can remove that, you shouldn't need negative margins on those elements, or the `right` and `bottom` you've added to the `.pf-c-drawer__splitter` element.

Reply:
@mcoker Thank you for you suggestions on `calc()` function, I will update it.
However, the reason I set `padding` and `margin` here is for the usage in React. When the `flex-basis` of the resizable panel is set to `0`, we need to use the `padding` to keep the splitter inside the drawer. Thus, we also need `margin` to keep the panel body in the right position. You may try to remove `padding` and `margin` mentioned above, but the splitter will be disappeared when you change the size of the panel to 0, then you will not be able to drag it back.
Or do you think there is a better way to keep the splitter inside the drawer?